tools/zetasql_helper/zetasql_helper/scanner/extract_function.cc

```diff
@@ -0,0 +1,94 @@
+//
+// Copyright 2020 BigQuery Utils
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "extract_function.h"
+#include "zetasql/public/analyzer.h"
+#include "zetasql/parser/parser.h"
+#include "zetasql_helper/util/util.h"
+
+namespace bigquery::utils::zetasql_helper {
+
+FunctionRange::FunctionRange(const zetasql::ASTFunctionCall *function_call) {
+  function = function_call->GetParseLocationRange();
+  name = function_call->function()->GetParseLocationRange();
+  std::vector<zetasql::ParseLocationRange> ranges;
+  for (auto argument : function_call->arguments()) {
+    ranges.push_back(argument->GetParseLocationRange());
+  }
+  arguments = ranges;
+}
+
+zetasql_base::StatusOr<FunctionRangeProto> FunctionRange::ToProto() const {
+  FunctionRangeProto proto;
+  // location range transfer
+  auto status_or_function_proto = function.ToProto();
+  ZETASQL_RETURN_IF_ERROR(status_or_function_proto.status());
+
+  auto range_proto = new zetasql::ParseLocationRangeProto(status_or_function_proto.value());
+  proto.set_allocated_function(range_proto);
+
+  // name range transfer
+  auto status_or_name_proto = name.ToProto();
+  ZETASQL_RETURN_IF_ERROR(status_or_name_proto.status());
+
+  range_proto = new zetasql::ParseLocationRangeProto(status_or_name_proto.value());
+  proto.set_allocated_name(range_proto);
+
+  for (auto &argument_range : arguments) {
```

Comment:
nit: const auto& argument_range: ....

Reply:
Just realized the formatter will concat the & and * with the variable..